app/services/declarations/create.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+module Declarations
+  class Create
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :lead_provider
+    attribute :participant_id
+    attribute :declaration_type, :string
+    attribute :declaration_date, :datetime
+    attribute :course_identifier, :string
+    attribute :has_passed
+
+    validates :lead_provider, presence: true
+    validates :participant_id, presence: { message: I18n.t("declaration.missing_participant_id") }
+    validates :participant, participant_presence: true, participant_not_withdrawn: true
+    validates :course_identifier, inclusion: { in: Course::IDENTIFIERS, message: I18n.t(:invalid_course) }, allow_blank: false, course_for_participant: true
+    validates :declaration_date, declaration_date: true, allow_blank: true
```

Comment:
Don't think we want to `allow_blank` for declaration date? even if its caught by the `declaration_date` validator its misleading having it here